src/processor.rs

```diff
@@ -208,4 +214,70 @@ impl Processor {
 
 		Ok(())
 	}
+
+	/// Cancel can be called after init_escrow. If called after exchange, it's already too late
+	/// since exchange is atomic and tokens have been transferred, so there will be no effect.
+	/// Since tokens haven't actually been transferred from initializer main token account to
+	/// initializer temp token account, we don't need to actually transfer any tokens.
+	/// What we need to do is:
+	/// 1) Close escrow info
+	/// 2) Close PDA account
```

Comment:
the enumeration of steps in the comment here is in the opposite order to what's being done in the actual implementation. Not a big deal, but still better to change the order to guide the reader's intuition.